### DISCUSSION TITLE
Inference Only Light Weight Module (without torchaudio)

### DISCUSSION THREAD

**Discussion by iprovalo** [coqui-ai/TTS#1687] [General Q&A] [answered]
I am trying to run TTS on Raspberry Pi.  Installing torch audio is a challenge.  The latest error I am getting is 

`"OSError: /usr/local/lib/python3.9/dist-packages/torchaudio/lib/libtorchaudio.so: undefined symbol: _ZNK5torch8autograd4Node4nameEv"`

when running inference with defaults:

`python3 TTS/bin/synthesize.py --text "Text for TTS"`

I was wondering how easily swappable  is the `torchaudio` dependency for the inference only use cases?

**Comment by p0p4k** [comment, marked as answer]
My 2 cents...
`torchaudio`  is mainly used for -
1) loading the audio files
2) resampling to desired sample rate based on the model requirements
3) compute melspectrogram
If you can replace those with some other library of your choice, it wont be an issue, I suppose. But, need to take care of datatypes (tensors, etc.)

**Reply by iprovalo** [reply]
Thank you, @p0p4k !

**Reply by p0p4k** [reply]
@iprovalo no problem. Good luck with your speech synthesis experiments!

**Comment by iprovalo** [comment]
It turns out that for the inference to work without torchaudio (in the current code base), I just need to address the mel-spectogram computation: 

1. Comment out [this  torchaudio.transforms.MelSpectrogram code](https://github.com/coqui-ai/TTS/blob/dev/TTS/encoder/models/base_encoder.py#L53-L60)
2. Comment back in [TorchSTFT code](https://github.com/coqui-ai/TTS/blob/dev/TTS/encoder/models/base_encoder.py#L38-L52)
3. Remove the import for [torchaudio](https://github.com/coqui-ai/TTS/blob/dev/TTS/encoder/models/base_encoder.py#L3)

**Reply by iprovalo** [reply]
@p0p4k I used this as a [reference](https://github.com/coqui-ai/TTS/commit/d39200e69b29b107302dcfc124a17faf0c0f6cdd).  

cc: @erogol @Edresson
